sha1.c fails to build with clang on OS X Lion

See https://trac.macports.org/ticket/51779 (which also includes a patch).

Building of gregorio with clang fails on Lion (more precisely, with Xcode >= 4.3 and < 5) with the following error:
```
gregorio-src/src/sha1.c:171:13: warning: implicit declaration of function 'alignof' is invalid in C99 [-Wimplicit-function-declaration]
        if (UNALIGNED_P(buffer)) {
            ^
gregorio-src/src/sha1.c:170:43: note: expanded from macro 'UNALIGNED_P'
#define UNALIGNED_P(p) ((uintptr_t) (p) % alignof (uint32_t) != 0)
                                          ^
gregorio-src/src/sha1.c:171:13: error: unexpected type name 'uint32_t': expected expression
        if (UNALIGNED_P(buffer)) {
            ^
gregorio-src/src/sha1.c:170:52: note: expanded from macro 'UNALIGNED_P'
#define UNALIGNED_P(p) ((uintptr_t) (p) % alignof (uint32_t) != 0)
                                                   ^
1 warning and 1 error generated.
```

--- a/configure.ac
+++ b/configure.ac
@@ -38,6 +38,8 @@ AC_PROG_YACC
 AC_CHECK_TOOL([RC], [windres], [no])
 AM_CONDITIONAL([HAVE_RC], [test x$RC != xno])
 
+AC_CHECK_ALIGNOF(uint32_t)
+
 AX_CHECK_COMPILE_FLAG([-std=gnu89], [CFLAGS+=" -std=gnu89"])
 dnl until flex gets their act together, use pedantic instead of pedantic-errors
 AX_CHECK_COMPILE_FLAG([-pedantic], [CFLAGS+=" -pedantic"])

--- a/src/sha1.c
+++ b/src/sha1.c
@@ -37,15 +37,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-#ifdef HAVE_STDALIGN_H
-#include <stdalign.h>
-#ifdef __clang__
-#pragma clang diagnostic ignored "-Wc11-extensions"
-#endif
-#else
-#define alignof(x) sizeof(x)
-#endif
-
 #ifdef WORDS_BIGENDIAN
 #define SWAP(n) (n)
 #else
@@ -167,7 +158,7 @@ void sha1_process_bytes(const void *buffer, size_t len, struct sha1_ctx *ctx)
     if (len >= 64) {
         /* architecture and data-specific; LCOV_EXCL_START */
 #if !_STRING_ARCH_unaligned
-#define UNALIGNED_P(p) ((uintptr_t) (p) % alignof (uint32_t) != 0)
+#define UNALIGNED_P(p) ((uintptr_t) (p) % ALIGNOF_UINT32_T != 0)
         if (UNALIGNED_P(buffer)) {
             while (len > 64) {
                 sha1_process_block(memcpy(ctx->buffer, buffer, 64), 64, ctx);

